apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/api/AndroidProject.kt

```diff
@@ -0,0 +1,76 @@
+package com.apollographql.apollo.gradle.api
+
+import com.android.build.gradle.AppExtension
+import com.android.build.gradle.BaseExtension
+import com.android.build.gradle.LibraryExtension
+import com.android.build.gradle.TestedExtension
+import com.android.build.gradle.api.BaseVariant
+import com.apollographql.apollo.gradle.internal.ApolloGenerateSourcesTask
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.file.Directory
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.TaskContainer
+import org.gradle.api.tasks.TaskProvider
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+object AndroidProject {
+  fun onEachVariant(project: Project, withTestVariants: Boolean = false, block: (BaseVariant) -> Unit) {
```

Comment:
> project is not used in this function. What about making this an extension over androidExtension or making androidExtension a parameter?

I like that it's somehow symmetric with `KotlinJvmProject` and `KotlinMultiplatformProject`. These helpers are now public APIs so users can use them to do all kind of crazy things if they really want to. They all take a `project` as a first parameter because this is what's usually available from `build.gradle[.kts]`